.github/workflows/ci.yml

```diff
@@ -0,0 +1,51 @@
+# Based on https://github.com/grafana/grafana-starter-datasource/blob/master/.github/workflows/ci.yml
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v2.1.2
+        with:
+          node-version: "14.x"
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - name: Cache yarn cache
+        uses: actions/cache@v2
+        id: cache-yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Cache node_modules
```

Comment:
is js dependency management really that broken that two caches are needed?

Reply:
I basically trust the Grafana starter template and hope that it does the right thing.
Perhaps it's some premature optimization, that is not strictly necessary, but speeds up the action for more complicated setups?

There seems to be a shortcut for setup-node that could be used: https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#user-content-caching-packages-data

https://github.com/actions/cache/blob/main/examples.md#node---npm
Hm. One should not cache the `node_modules` folder at all. So I will have to change this.

Reply:
I simplified the github action. Thanks for bringing it to my attention!